packages/edit-site/src/components/page-media/index.js

```diff
@@ -0,0 +1,203 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	Icon,
+	__experimentalHeading as Heading,
+	__experimentalHStack as HStack,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useEntityRecords } from '@wordpress/core-data';
+import { decodeEntities } from '@wordpress/html-entities';
+import { useState, useMemo, useCallback } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import Page from '../page';
+import { DataViews } from '../dataviews';
+import { useAddedBy, AvatarImage } from '../list/added-by';
+
+const EMPTY_ARRAY = [];
+
+const DEFAULT_VIEW = {
+	type: 'list',
+	search: '',
+	page: 1,
+	perPage: 20,
+	// All fields are visible by default, so it's
+	// better to keep track of the hidden ones.
+	hiddenFields: [],
+	layout: {},
+};
+
+function AuthorField( { item } ) {
+	const { text, icon, imageUrl } = useAddedBy( item.type, item.id );
+	return (
+		<HStack alignment="left">
+			{ imageUrl ? (
+				<AvatarImage imageUrl={ imageUrl } />
+			) : (
+				<div className="edit-site-list-added-by__icon">
+					<Icon icon={ icon } />
+				</div>
+			) }
+			<span>{ text }</span>
+		</HStack>
+	);
+}
+
+export default function PageMedia() {
+	const [ view, setView ] = useState( DEFAULT_VIEW );
+	const { records: attachments, isResolving: isLoadingData } =
+		useEntityRecords( 'root', 'media', {
+			per_page: -1,
+		} );
+
+	const mediaItems = useMemo( () => {
+		if ( ! attachments ) {
+			return EMPTY_ARRAY;
+		}
+		return attachments.map( ( item ) => ( {
+			title: item.title?.rendered || item.slug || __( '(no title)' ),
+			type: item.media_type,
+			alt: item.alt_text,
+			thumbnail: item.media_details.sizes.thumbnail.source_url,
```

Comment:
Totally optional (and not sure how much optional chaining is required), but I need to add some optional chaining to get the page to load (something complained about `.source_url` not being present in my environment) — could be the non image files I have in my library 🤔

```suggestion
			thumbnail: item?.media_details?.sizes?.thumbnail?.source_url,
```

Reply:
Ah, I haven't tested this branch at all so don't expect much from it 🤣 

I'll add more media items to my library and fix as I go.

Thanks for checking it out already though!